crates/filter-parser/src/lib.rs

```diff
@@ -136,13 +139,23 @@ impl<'a> From<&'a str> for Token<'a> {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum VectorFilter<'a> {
+    Fragment(Token<'a>),
+    DocumentTemplate,
+    UserProvided,
+    Regenerate,
+    None,
+}
+
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub enum FilterCondition<'a> {
     Not(Box<Self>),
     Condition { fid: Token<'a>, op: Condition<'a> },
     In { fid: Token<'a>, els: Vec<Token<'a>> },
     Or(Vec<Self>),
     And(Vec<Self>),
+    VectorExists { fid: Token<'a>, embedder: Option<Token<'a>>, filter: VectorFilter<'a> },
```

Comment:
do we use the `fid` in this variant? Like in error messages or the like?

Reply:
Yes [here](https://github.com/meilisearch/meilisearch/blob/6c3dd83ae528a8b82614880c0206cd7886a187c7/crates/filter-parser/src/lib.rs#L206). It's only used for the message to display when trying to use vector filters without enabling the multimodal feature. We will be able to remove it in the future